schematic/store/synapse.py

```diff
@@ -441,11 +474,112 @@ def get_synapse_table(self, synapse_id: str) -> Tuple[pd.DataFrame, CsvFileTable
+        # this could create a divergence between manifest column and annotations. this should be ok for most use cases.
+        # columns with special characters are outside of the schema
+        metadataSyn = {}
+        blacklist_chars = ['(', ')', '.', ' ']
+        
+        for k, v in row.to_dict().items():
+
+            if useSchemaLabel:
+                keySyn = se.get_class_label_from_display_name(str(k)).translate({ord(x): '' for x in blacklist_chars})
+            else:
+                keySyn = str(k)
+
+            # Skip `Filename` and `ETag` columns when setting annotations
+            if keySyn in ["Filename", "ETag", "eTag"]:
+                continue
+
+            # truncate annotation values to 500 characters if the
+            # size of values is greater than equal to 500 characters
+            # add an explicit [truncatedByDataCuratorApp] message at the end
+            # of every truncated message to indicate that the cell value
+            # has been truncated
+            if isinstance(v, str) and len(v) >= 500:
+                v = v[0:472] + "[truncatedByDataCuratorApp]"
+
+            metadataSyn[keySyn] = v
+
+        # set annotation(s) for the various objects/items in a dataset on Synapse
+        annos = self.syn.get_annotations(entityId)
+
+        for anno_k, anno_v in metadataSyn.items():
+            
+            #Do not save blank annotations as NaNs,
+            #remove keys with nan/blank values from dict of annotations to be uploaded if present on current data annotation
+            if isinstance(anno_v,float) and np.isnan(anno_v):
+                if hideBlanks:
+                    annos.pop(anno_k) if anno_k in annos.keys() else annos
+                else:
+                    annos[anno_k] = ""
+            else:
+                annos[anno_k] = anno_v
+        return annos
+
     def associateMetadataWithFiles(
-        self, metadataManifestPath: str, datasetId: str, useSchemaLabel: bool = True, hideBlanks: bool = False,
+        self, metadataManifestPath: str, datasetId: str, manifest_record_type: str, 
+        useSchemaLabel: bool = True, hideBlanks: bool = False,
     ) -> str:
         """Associate metadata with files in a storage dataset already on Synapse.
         Upload metadataManifest in the storage dataset folder on Synapse as well. Return synapseId of the uploaded manifest file.
+        
+        If this is a new manifest there could be no Synapse entities associated with the rows of this manifest
+        this may be due to data type (e.g. clinical data) being tabular
+        and not requiring files; to utilize uniform interfaces downstream
+        (i.e. fileviews), a Synapse entity (a folder) is created for each row
+        and an entity column is added to the manifest containing the resulting
+        entity IDs; a table is also created at present as an additional interface
+        for downstream query and interaction with the data.
```

Comment:
This now applies only if they have indicated manifest_record_type is 'entity'?

Reply:
How does this update sound?
``` 
    If this is a new manifest there could be no Synapse entities associated with the rows of this manifest
    this may be due to data type (e.g. clinical data) being tabular
    and not requiring files; If manifest_record_type is 'entity' or 'both', to utilize uniform interfaces downstream
    (i.e. fileviews), a Synapse entity (a folder) is created for each row
    and an entity column is added to the manifest containing the resulting
    entity IDs; a table is also created at present as an additional interface
    for downstream query and interaction with the data. Otherwise, if
    manifest record type is 'table' a the manifest will be stored as a table, that
    can be joined with fileviews through materialized views. The 'entityId' column will
    contain the synapseId of the generated table in this case.```

Reply:
How does this update sound?
``` 
    If this is a new manifest there could be no Synapse entities associated with the rows of this manifest
    this may be due to data type (e.g. clinical data) being tabular
    and not requiring files; If manifest_record_type is 'entity' or 'both', to utilize uniform interfaces downstream
    (i.e. fileviews), a Synapse entity (a folder) is created for each row
    and an entity column is added to the manifest containing the resulting
    entity IDs; a table is also created at present as an additional interface
    for downstream query and interaction with the data. Otherwise, if
    manifest record type is 'table' a the manifest will be stored as a table, that
    can be joined with fileviews through materialized views. The 'entityId' column will
    contain the synapseId of the generated table in this case.```